### DISCUSSION TITLE
Proxy Host > HTTP only to Windows Web Server > Certify the Web

### DISCUSSION THREAD

**Discussion by maieredv-manuel** [ZoeyVid/NPMplus#1019] [Q&A] [answered]
Hello again :-)

after switching from "Classic" NPM to NPMPlus i have a second strange Problem.
I have 3 Proxyhosts set with HTTP Only that are pointing directly to one of my Servers.
On that Server are running some Services that arent direktly HTTP related (SMTP Server for example).
But there Services also need Certificates, so i order and renew them via Certify the  Web.

Now i have on one of those Hosts the problem that the renewal isnt working anymore, when i navigate to the Path were the Files are places, i get an Error 404 from NPMPlus and not from the Server itselfs.

**Comment by Zoey2936** [comment]
what path do you mean?

**Comment by maieredv-manuel** [comment]
Ah, sorry ....
domainname.de/.well-known/acme-challenge 
/.well-known connects to the right Server 
 /.well-known/acme-challenge comes back from NPMPlus

**Comment by Zoey2936** [comment, marked as answer]
yes, they are used by npmplus, otherwise certbot would not work (it seems like it is broken in vanilla npm, thats why it is working for you on vanilla, normally it wouldn't work there too)

**Comment by maieredv-manuel** [comment]
Aaaah OK :-( 
But then there is also an Bug in NPMPlus, two of three requests are without problems.
I mean for HTTPs Hosts its OK - there isnt any need to get the Certs on the upstream host, but on a HTTP only host, there is no need to "grab" the acme-challenge path!?

Is there a simple way to the the Certs out of NPMPlus? 
Download with WinSCP is a bit tricky ... i think about something like a simple Download ;-)

**Reply by Zoey2936** [reply]
you can generate the certs with npmplus and download them?

**Reply by maieredv-manuel** [reply]
Yeah sure, but not via script / automated.
Id like to download the Certs with a script and import them to the service where its needed.

**Reply by Zoey2936** [reply]
if they are on the same server, you can copy the certs using cron?

**Reply by maieredv-manuel** [reply]
saadly not, the services are Windows based.
I think its the easyiest way to download them with a WinSCP Script :-)

Thank you for your work!

**Reply by Zoey2936** [reply]
maybe you could use dns based acme cert generation on your windows server?

**Reply by maieredv-manuel** [reply]
Hmm, saadly my Domainhoster dont offers and automated way to set the right TXT Entrys 👎 
I think ill go the WinSCP way - i can connect to the Docker Host via SSH so i only need the right commands :-)
